src/java.base/share/classes/jdk/internal/util/Exceptions.java

```diff
@@ -0,0 +1,322 @@
+        }
+        @Override
+        public String output() {
+            setup();
+            return super.output(enhancedNonSocketExceptionText);
+        }
+    }
+
+    static final class JarInfo extends SensitiveInfo {
+        public JarInfo(String name) {
+             super(name);
+        }
+        @Override
+        public String output() {
+            setup();
+            return super.output(enhancedJarExceptionText);
+        }
+    }
+
+    static final class UserInfo extends SensitiveInfo {
+        public UserInfo(String host) {
+             super(host);
+        }
+        @Override
+        public String output() {
+            setup();
+            return super.output(enhancedUserExceptionText);
+        }
+    }
+
+    // remove leading, trailing and duplicated space characters
+    static String trim(String s) {
+        int len = s.length();
+        if (len == 0) return s;
+
+        StringBuilder sb = new StringBuilder();
+
+        // initial value deals with leading spaces
+        boolean inSpace = true;
+        for (int i=0; i<len; i++) {
+            char c = s.charAt(i);
+            if (c == ' ') {
+                if (inSpace)
+                    continue;
+                inSpace = true;
+            } else
+                inSpace = false;
+            sb.append(c);
+        }
+        int sblen = sb.length();
+        // last char could be a space
+        if (sblen > 0 && sb.charAt(sblen - 1) == ' ')
+            sb.deleteCharAt(sblen - 1);
+        return sb.toString();
+    }
+
+    public static SensitiveInfo filterSocketInfo(String host) {
+        return new SocketInfo(host);
+    }
+
+    public static SensitiveInfo filterNonSocketInfo(String host) {
+        return new NonSocketInfo(host);
+    }
+
+    public static SensitiveInfo filterJarName(String name) {
+        return new JarInfo(name);
+    }
+
+    public static SensitiveInfo filterUserName(String name) {
+        return new UserInfo(name);
+    }
+
+    /**
+     * Transform each SensitiveInfo into a String argument which is passed
+     * to String.format(). This string is then trimmed.
+     */
+    public static String formatMsg(String format, SensitiveInfo... infos) {
+        String[] args = new String[infos.length];
+
+        int i = 0;
+
+        for (SensitiveInfo info : infos) {
+            args[i++] = info.output();
+        }
+        return trim(String.format(format, (Object[])args));
+    }
+
+    /**
+     * Simplification of above. Equivalent to:
+     *       formatMsg("%s", SensitiveInfo[1]); // ie with one arg
+     */
+    private static String formatMsg(SensitiveInfo info) {
+        return trim(info.output());
+    }
+
+    public static void setup() {
+        if (initialized || !VM.isBooted())
+            return;
+        enhancedSocketExceptionText = SecurityProperties.includedInExceptions("hostInfo");
+        enhancedNonSocketExceptionText = SecurityProperties.includedInExceptions("hostInfoExclSocket")
```

Comment:
This looks like the inverse of the previous use of a socket category, except this time it anything that not in Socket.

Consider the following: 
includeInException specifies the type of information that maybe included in an enhanced exception 
e.g. Hostname, IPAddress, PortNumber, UserDetails, Uri (including Urls), JarDetails, All
This defines an information policy 

developer are familiar with packages, so a second property specified the "domain" of application of an information policy:  enhancedException.packages specifies a list of packages where the includeInException information policy will apply

an empty list or the enhancedException.packages means freedom of information and the defined includedInException applies to all packages

Reply:
The point I was raising, is that the socket etc category was dropped under the premise that it requires knowledge of the APIs used. 
But equally the setting hostInfoExclSocket suggests some knowledge of APIs is available, so there is a contradiction in the rationale presented above
Also the setting enhancedNonSocketExceptionText is inclusive of the setting enhancedSocketExceptionText, which adds further subtleties to the configuration, which was previously suggested should be avoided.